src/consensus/aft/async_execution.h

```diff
@@ -301,4 +303,30 @@ namespace aft
     size_t size;
     OArray oarray;
   };
+
+  class NodeCatchUpCallback : public AbstractMsgCallback
+  {
+  public:
+    NodeCatchUpCallback(
```

Comment:
This is not following the pattern of the rest of the code. You can add a virtual method to store update_node_progress